examples/weather_station/src/app_task.cpp

```diff
@@ -339,6 +363,121 @@ void AppTask::UpdateClusterState()
 	}
 }
 
+void AppTask::UpdatePowerSourceClusterState() {
+	EmberAfStatus status;
+	uint32_t voltage = BatteryMeasurementRead();
+	/* Value is expressed in half percent units ranging from 0 to 200. */
+	uint8_t batteryPercentage;
+	EmberAfPowerSourceStatus batteryStatus;
+	EmberAfBatChargeLevel batteryChargeLevel;
+	bool batteryPresent;
+	EmberAfBatChargeState batteryCharged;
+
+	if (voltage < 0) {
+		voltage = 0;
+		batteryPercentage = 0;
+		batteryStatus = EMBER_ZCL_POWER_SOURCE_STATUS_UNAVAILABLE;
+		batteryPresent = false;
+
+		LOG_ERR("Battery level measurement failed %d", voltage);
+
+	} else {
+		batteryStatus = EMBER_ZCL_POWER_SOURCE_STATUS_ACTIVE;
+		batteryPresent = true;
+	}
+
+	if (voltage <= kMinimalOperatingVoltageMv) {
+		batteryPercentage = kMinBatteryPercentage;
+	} else if (voltage >= kMaximalOperatingVoltageMv) {
+		batteryPercentage = kMaxBatteryPercentage;
+	} else {
+		batteryPercentage = kMaxBatteryPercentage * (voltage - kMinimalOperatingVoltageMv) / (kMaximalOperatingVoltageMv - kMinimalOperatingVoltageMv);
+	}
+
+	if ((voltage < kWarningThresholdVoltageMv) && (voltage >= kCriticalThresholdVoltageMv)) {
+		batteryChargeLevel = EMBER_ZCL_BAT_CHARGE_LEVEL_WARNING;
+	} else if (voltage < kCriticalThresholdVoltageMv) {
+		batteryChargeLevel = EMBER_ZCL_BAT_CHARGE_LEVEL_CRITICAL;
+	} else {
+		batteryChargeLevel = EMBER_ZCL_BAT_CHARGE_LEVEL_OK;
+	}
+
+	if (BatteryCharged()) {
+		batteryCharged = EMBER_ZCL_BAT_CHARGE_STATE_IS_CHARGING;
+	} else {
+		batteryCharged = EMBER_ZCL_BAT_CHARGE_STATE_IS_NOT_CHARGING;
+	}
+
+	status = emberAfWriteAttribute(kPowerSourceEndpointId,
+				       ZCL_POWER_SOURCE_CLUSTER_ID,
+					   ZCL_POWER_SOURCE_BAT_VOLTAGE_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+				       reinterpret_cast<uint8_t *>(&voltage), ZCL_INT32U_ATTRIBUTE_TYPE);
+
+	if (status != EMBER_ZCL_STATUS_SUCCESS) {
+		LOG_ERR("Updating battery voltage failed %x", status);
+	}
+
+	status = emberAfWriteAttribute(kPowerSourceEndpointId,
+				       ZCL_POWER_SOURCE_CLUSTER_ID,
+					   ZCL_POWER_SOURCE_BAT_PERCENT_REMAINING_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+				       reinterpret_cast<uint8_t *>(&batteryPercentage), ZCL_INT8U_ATTRIBUTE_TYPE);
+
+	if (status != EMBER_ZCL_STATUS_SUCCESS) {
+		LOG_ERR("Updating battery percentage failed %x", status);
+	}
+
+	status = emberAfWriteAttribute(kPowerSourceEndpointId,
```

Comment:
The current approach seems to be using `PowerSource::Attributes::BatteryChargeLevel::Set` etc. accessors from `Accessors.h` file. Could you use it, instead of the old API?